### ISSUE TITLE
UnsatisfiableError sometimes leads to TypeError: sequence item 0: expected str instance, Channel found

### ISSUE THREAD

**Issue by gerritholl** [conda/conda#9681]
## Current Behavior

Trying to create a new environment using Python 3.8 and a package from the `rttools` conda channel, sometimes results in a traceback with a `TypeError: sequence item 0: expected str instance, Channel found`.  Sometimes it runs correctly and reports an `UnsatisfiableError` gracefully.

### Steps to Reproduce

```bash
conda create -n test -c rttools -c conda-forge --strict-channel-priority python=3.8 typhon
```

Sometimes this results in:

```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/exceptions.py", line 1079, in __call__
        return func(*args, **kwargs)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        return getattr(module, func_name)(args, parser)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/cli/main_create.py", line 37, in execute
        install(args, parser, 'create')
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/cli/install.py", line 261, in install
        unlink_link_transaction = solver.solve_for_transaction(
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 114, in solve_for_transaction
        unlink_precs, link_precs = self.solve_for_diff(update_modifier, deps_modifier,
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 157, in solve_for_diff
        final_precs = self.solve_final_state(update_modifier, deps_modifier, prune, ignore_pinned,
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 281, in solve_final_state
        ssc = self._run_sat(ssc)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 805, in _run_sat
        ssc.solution_precs = ssc.r.solve(tuple(final_environment_specs),
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 1318, in solve
        self.find_conflicts(specs, specs_to_add, history_specs)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 347, in find_conflicts
        bad_deps = self.build_conflict_map(specs, specs_to_add, history_specs)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 506, in build_conflict_map
        c = self.breadth_first_search_for_dep_graph(
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 369, in breadth_first_search_for_dep_graph
        last_spec = MatchSpec.union((path[-1], target_paths[-1][-1]))[0]
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 481, in union
        return cls.merge(match_specs, union=True)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 475, in merge
        reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 475, in <lambda>
        reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 502, in _merge
        final = this_component.union(that_component)
      File "/data/gholl/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 764, in union
        return '|'.join(options)
    TypeError: sequence item 0: expected str instance, Channel found

`$ /data/gholl/miniconda3/bin/conda create -n test -c rttools -c conda-forge --strict-channel-priority python=3.8 typhon`

  environment variables:
         ALSA_CONFIG_PATH=/etc/alsa-pulse.conf
                 CIO_TEST=<not set>
                CONDA_EXE=/data/gholl/miniconda3/bin/conda
         CONDA_PYTHON_EXE=/data/gholl/miniconda3/bin/python
               CONDA_ROOT=/data/gholl/miniconda3
              CONDA_SHLVL=0
                  MANPATH=/media/nas/x21308/inst/texlive/2019/texmf-
                          dist/doc/man:/media/nas/x21308/inst/texlive/2019/texmf-
                          dist/doc/man:/media/nas/x21308/inst/texlive/2019/texmf-
                          dist/doc/man:/usr/local/man:/usr/share/man
                 NO_PROXY=<set>
                     PATH=/data/gholl/miniconda3/bin:/media/nas/x21308/inst/texlive/2019/bin/x86
                          _64-linux:/media/nas/x21308/inst/texlive/2019/bin/x86_64-linux:/media/
                          nas/x21308/inst/texlive/2019/bin/x86_64-linux:/media/nas/x21324/minico
                          nda3/condabin:/sbin:/usr/sbin:/usr/local/sbin:/root/bin:/usr/local/bin
                          :/usr/bin:/bin:/opt/bin:/usr/lib/mit/sbin:/opt/perforce/bin:/opt/hdf-
                          java/bin:/usr/lib/mit/sbin:/opt/perforce/bin
            PYTHONSTARTUP=/home/gholl/.pythonrc.py
       REQUESTS_CA_BUNDLE=<not set>
              SOCKS_PROXY=<set>
            SSL_CERT_FILE=<not set>
               WINDOWPATH=7
                 XNLSPATH=/usr/share/X11/nls
                ftp_proxy=<set>
             gopher_proxy=<set>
               http_proxy=<set>
              https_proxy=<set>
                 no_proxy=<set>
              socks_proxy=<set>

     active environment : None
            shell level : 0
       user config file : /home/gholl/.condarc
 populated config files : /home/gholl/.condarc
          conda version : 4.8.2
    conda-build version : not installed
         python version : 3.8.1.final.0
       virtual packages : __cuda=10.0
                          __glibc=2.26
       base environment : /data/gholl/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/rttools/linux-64
                          https://conda.anaconda.org/rttools/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /data/gholl/miniconda3/pkgs
                          /home/gholl/.conda/pkgs
       envs directories : /data/gholl/miniconda3/envs
                          /home/gholl/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.2 requests/2.22.0 CPython/3.8.1 Linux/4.12.14-lp150.12.82-default opensuse-leap/15.0 glibc/2.26
                UID:GID : 16209:16217
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.

If submitted, this report will be used by core maintainers to improve
future releases of conda.
Would you like conda to send this report to the core maintainers?

[y/N]:

No report sent. To permanently opt-out, use

    $ conda config --set report_errors false
```

Sometimes conda doesn't throw an exception, but instead outputs correctly that there are unsatisfiable specifications:

```
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... 
Building graph of deps:   0%|          | 0/4 [00:00<?, ?it/s]
Examining @/linux-64::__cuda==10.0=0:   0%|          | 0/4 [00:00<?, ?it/s]
Examining typhon:  25%|██▌       | 1/4 [00:00<00:00, 4315.13it/s]          
Examining python=3.8:  50%|█████     | 2/4 [00:02<00:02,  1.36s/it]
Examining python=3.8:  75%|███████▌  | 3/4 [00:02<00:00,  1.10it/s]
Examining @/linux-64::__glibc==2.26=0:  75%|███████▌  | 3/4 [00:02<00:00,  1.10it/s]
                                                                                    

Determining conflicts:   0%|          | 0/4 [00:00<?, ?it/s]
Examining conflict for __cuda:   0%|          | 0/4 [00:00<?, ?it/s]
Examining conflict for typhon python:  25%|██▌       | 1/4 [00:00<00:00, 10699.76it/s]
                                                                                      
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package python conflicts for:
typhon -> python[version='3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
typhon -> docutils -> python[version='2.7.*|>=2.7,<2.8.0a0|>=3.8,<3.9.0a0|3.4.*|>=3.6|>=3.5.3|>=3.5|>=3.8.0a,<3.9.0a0']
python=3.8The following specifications were found to be incompatible with your CUDA driver:

  - feature:/linux-64::__cuda==10.0=0

Your installed CUDA driver is: 10.0

Note that strict channel priority may have removed packages required for satisfiability.
```

## Expected Behavior

I think that if there are unsatisfiable conditions, it should report that there are unsatisfiable conditions, like in the second case, and not (sometimes) throw me a long traceback with "An unexpected error occured"

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : None
            shell level : 0
       user config file : /home/gholl/.condarc
 populated config files : /home/gholl/.condarc
          conda version : 4.8.2
    conda-build version : not installed
         python version : 3.8.1.final.0
       virtual packages : __cuda=10.0
                          __glibc=2.26
       base environment : /data/gholl/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/rttools/linux-64
                          https://conda.anaconda.org/rttools/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /data/gholl/miniconda3/pkgs
                          /home/gholl/.conda/pkgs
       envs directories : /data/gholl/miniconda3/envs
                          /home/gholl/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.2 requests/2.22.0 CPython/3.8.1 Linux/4.12.14-lp150.12.82-default opensuse-leap/15.0 glibc/2.26
                UID:GID : 16209:16217
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /home/gholl/.condarc <==
channel_priority: strict
channels:
  - rttools
  - conda-forge
  - defaults
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /data/gholl/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
asn1crypto                1.3.0                    py38_0    defaults
ca-certificates           2020.1.1                      0    defaults
certifi                   2019.11.28               py38_0    defaults
cffi                      1.14.0           py38h2e261b9_0    defaults
chardet                   3.0.4                 py38_1003    defaults
conda                     4.8.2                    py38_0    defaults
conda-env                 2.6.0                         1    defaults
conda-package-handling    1.6.0            py38h7b6447c_0    defaults
cryptography              2.8              py38h1ba5d50_0    defaults
idna                      2.8                   py38_1000    defaults
ld_impl_linux-64          2.33.1               h53a641e_7    defaults
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 9.2.0                hdf63c60_0    conda-forge
libstdcxx-ng              8.2.0                hdf63c60_1    defaults
ncurses                   6.1                  he6710b0_1    defaults
openssl                   1.1.1d               h7b6447c_4    defaults
pip                       20.0.2                   py38_1    defaults
pycosat                   0.6.3            py38h7b6447c_0    defaults
pycparser                 2.19                       py_0    defaults
pyopenssl                 19.1.0                   py38_0    defaults
pysocks                   1.7.1                    py38_0    defaults
python                    3.8.1                h357f687_2    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
requests                  2.22.0                   py38_1    defaults
ruamel_yaml               0.15.80         py38h516909a_1000    conda-forge
setuptools                45.2.0                   py38_0    defaults
six                       1.14.0                   py38_0    defaults
sqlite                    3.31.1               h7b6447c_0    defaults
tk                        8.6.10               hed695b0_0    conda-forge
tqdm                      4.42.1                     py_0    defaults
urllib3                   1.25.8                   py38_0    defaults
wheel                     0.34.2                   py38_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.2.2                h516909a_1    conda-forge
zlib                      1.2.11               h7b6447c_3    defaults
```
</p></details>

**Comment by markm00001** [comment]
### I have been struggling with the same error message mentioned above since I upgraded to conda 4.8.2. when I try to install (or update) just about any package (numpy, pandas, etc) into a environment containing python 3.6.1.

(testpkgs) km@algobox:~$ conda install pandas==0.24.2
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: / 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                         

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/cli/install.py", line 261, in install
        unlink_link_transaction = solver.solve_for_transaction(
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 114, in solve_for_transaction
        unlink_precs, link_precs = self.solve_for_diff(update_modifier, deps_modifier,
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 157, in solve_for_diff
        final_precs = self.solve_final_state(update_modifier, deps_modifier, prune, ignore_pinned,
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 281, in solve_final_state
        ssc = self._run_sat(ssc)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 805, in _run_sat
        ssc.solution_precs = ssc.r.solve(tuple(final_environment_specs),
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 1316, in solve
        raise UnsatisfiableError({})
    conda.exceptions.UnsatisfiableError: 
    Did not find conflicting dependencies. If you would like to know which
    packages conflict ensure that you have enabled unsatisfiable hints.
    
    conda config --set unsatisfiable_hints True
                
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/exceptions.py", line 1079, in __call__
        return func(*args, **kwargs)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        return getattr(module, func_name)(args, parser)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/cli/install.py", line 295, in install
        unlink_link_transaction = solver.solve_for_transaction(
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 114, in solve_for_transaction
        unlink_precs, link_precs = self.solve_for_diff(update_modifier, deps_modifier,
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 157, in solve_for_diff
        final_precs = self.solve_final_state(update_modifier, deps_modifier, prune, ignore_pinned,
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 281, in solve_final_state
        ssc = self._run_sat(ssc)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/core/solve.py", line 805, in _run_sat
        ssc.solution_precs = ssc.r.solve(tuple(final_environment_specs),
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 1318, in solve
        self.find_conflicts(specs, specs_to_add, history_specs)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 347, in find_conflicts
        bad_deps = self.build_conflict_map(specs, specs_to_add, history_specs)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 506, in build_conflict_map
        c = self.breadth_first_search_for_dep_graph(
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/resolve.py", line 369, in breadth_first_search_for_dep_graph
        last_spec = MatchSpec.union((path[-1], target_paths[-1][-1]))[0]
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 481, in union
        return cls.merge(match_specs, union=True)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 475, in merge
        reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 475, in <lambda>
        reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 502, in _merge
        final = this_component.union(that_component)
      File "/home/km/miniconda3/lib/python3.8/site-packages/conda/models/match_spec.py", line 764, in union
        return '|'.join(options)
    TypeError: sequence item 0: expected str instance, Channel found

`$ /home/km/miniconda3/bin/conda install pandas==0.24.2`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=testpkgs
                CONDA_EXE=/home/km/miniconda3/bin/conda
             CONDA_PREFIX=/home/km/miniconda3/envs/testpkgs
    CONDA_PROMPT_MODIFIER=(testpkgs)
         CONDA_PYTHON_EXE=/home/km/miniconda3/bin/python
               CONDA_ROOT=/home/km/miniconda3
              CONDA_SHLVL=1
                     PATH=/home/km/miniconda3/bin:/home/km/miniconda3/envs/testpkgs/bin:/home/km
                          /miniconda3/condabin:/home/km/.local/bin:/home/km/miniconda3/bin:/usr/
                          local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/us
                          r/local/games:/snap/bin
               PYTHONPATH=:/home/km/pyProjects
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : testpkgs
    active env location : /home/km/miniconda3/envs/testpkgs
            shell level : 1
       user config file : /home/km/.condarc
 populated config files : /home/km/.condarc
          conda version : 4.8.2
    conda-build version : not installed
         python version : 3.8.1.final.0
       virtual packages : __glibc=2.27
       base environment : /home/km/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/km/miniconda3/pkgs
                          /home/km/.conda/pkgs
       envs directories : /home/km/miniconda3/envs
                          /home/km/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.2 requests/2.22.0 CPython/3.8.1 Linux/5.0.0-1029-gcp ubuntu/18.04.1 glibc/2.27
                UID:GID : 1002:1003
             netrc file : None
           offline mode : False

**Comment by ngoldbaum** [comment]
I can also trigger this by doing `conda create -q -n test-environment pytorch cpuonly torchvision python=3.7 -c pytorch-nightly`.

**Comment by sh1ng** [comment]
The same when use conda 4.8.2 and python 3.6

**Comment by astraw** [comment]
#9748 seems related and was merged and released in 4.8.3.

**Comment by unicrus** [comment]
Upgrade to conda 4.8.3 solved the issue in my case.

**Comment by muyue1238** [comment]
I'm using conda4.8.3 and the problem exists.

**Comment by stephenkraemer** [comment]
in my hands this problem presented itself in different situations due to different reasons. One case that was not solved by upgrading to 4.8.3 was in the end simply related to an accidental switch of the *channel order* (conda-forge vs bioconda) when porting an environment to a new machine. So the general symptom may also just be caused by a 'user-specific' problem in the conda config.
